tests/taproot_maker_abort2.rs

```diff
@@ -0,0 +1,210 @@
+        maker.wallet().write().unwrap().sync_and_save().unwrap();
+    }
+    let actual_maker_spendable_balances = verify_maker_pre_swap_balance_taproot(&taproot_makers);
+    info!("🔄 Initiating taproot maker abort 2 case. (Will fail mid-swap due to one maker closing connection at handling private key step)");
+
+    // Swap params - small amount for faster testing
+    let swap_params = SwapParams {
+        send_amount: Amount::from_sat(500000), // 0.005 BTC
+        maker_count: 2,
+        tx_count: 3,
+        required_confirms: 1,
+        manually_selected_outpoints: None,
+    };
+
+    // Attempt the swap - it will fail when maker closes connection
+    // After recovery, do_coinswap returns Ok(None) to indicate recovery was triggered
+    match taproot_taker.do_coinswap(swap_params) {
+        Ok(Some(_report)) => {
+            panic!("Swap should have failed due to maker closing connection, but succeeded with report!");
+        }
+        Ok(None) => {
+            info!("✅ Taproot coinswap triggered recovery as expected (Ok(None))");
+        }
+        Err(e) => {
+            info!("✅ Taproot coinswap failed as expected: {:?}", e);
+        }
+    }
+
+    // Mine blocks to confirm any recovery transactions
+    info!("⛏️ Mining blocks to confirm recovery transactions...");
+    generate_blocks(bitcoind, 2);
+    taproot_taker.get_wallet_mut().sync_and_save().unwrap();
+
+    info!("📊 Taproot Taker balance after hashlock recovery:");
+    let taker_balances_after = taproot_taker.get_wallet().get_balances().unwrap();
+    info!(
+        "  Regular: {}, Contract: {}, Spendable: {}",
+        taker_balances_after.regular, taker_balances_after.contract, taker_balances_after.spendable
+    );
+
+    // Now wait for maker to extract preimage and recover via hashlock
+    info!("⏳ Waiting for maker to extract preimage and recover via hashlock...");
+    std::thread::sleep(std::time::Duration::from_secs(60));
+    // Mine more blocks to give maker time to see the hashlock sweep
+    generate_blocks(bitcoind, 2);
+
+    // Verify swap results
+    let taker_wallet = taproot_taker.get_wallet();
+    let taker_balances = taker_wallet.get_balances().unwrap();
+
+    // Use spendable balance (regular + swap) since swept coins from V2 swaps
+    // are tracked as SweptCoinV2 and appear in swap balance
+    // Here in hashlock recovery the spendable balance is almost similar to key-path spend
+    // as the parties are completing their swap by claiming their incoming contract
+    // via script-path spend.
+    let taker_total_after = taker_balances.spendable;
+    assert!(
+        taker_total_after.to_sat() == 14944003,
+        "Taproot Taker Balance check after hashlock recovery. Original: {}, After: {}",
+        taproot_taker_original_balance,
+        taker_total_after
+    );
+
+    let balance_diff = taproot_taker_original_balance - taker_total_after;
+    assert!(
+        balance_diff.to_sat() == 55997, // Hashlock recovery fee
```

Comment:
This is not really just the hashlock fee. This is maker_fee + hashlock_tx_fee. Would be good to separate these two out. Just assert it as sum of those two numbers and explain in the comment.